src/BizHawk.Emulation.Cores/Computers/DOS/DOSBox.cs

```diff
@@ -0,0 +1,492 @@
+			fi.joystick2.right   = _syncSettings.EnableJoystick2 && controller.IsPressed($"P2 {Inputs.Joystick} {JoystickButtons.Right}") ? 1 : 0;
+			fi.joystick2.button1 = _syncSettings.EnableJoystick2 && controller.IsPressed($"P2 {Inputs.Joystick} {JoystickButtons.Button1}") ? 1 : 0;
+			fi.joystick2.button2 = _syncSettings.EnableJoystick2 && controller.IsPressed($"P2 {Inputs.Joystick} {JoystickButtons.Button2}") ? 1 : 0;
+
+			// Setting mouse inputs
+			if (_syncSettings.EnableMouse)
+			{
+				// 272 is minimal delta for RMouse on my machine, this will be obsolete when global sensitivity for RMouse is added and when it's bindable from GUI
+				var deltaX = controller.AxisValue($"{Inputs.Mouse} {MouseInputs.XDelta}") / 272;
+				var deltaY = controller.AxisValue($"{Inputs.Mouse} {MouseInputs.YDelta}") / 272;
+				fi.mouse.posX = controller.AxisValue($"{Inputs.Mouse} {MouseInputs.XAxis}");
+				fi.mouse.posY = controller.AxisValue($"{Inputs.Mouse} { MouseInputs.YAxis}");
+				fi.mouse.dX = deltaX != 0 ? deltaX : fi.mouse.posX - _mouseState.posX;
+				fi.mouse.dY = deltaY != 0 ? deltaY : fi.mouse.posY - _mouseState.posY;
+
+				// Button pressed criteria:
+				// If the input is made in this frame and the button is not held from before
+				fi.mouse.leftButtonPressed    = controller.IsPressed($"{Inputs.Mouse} {MouseInputs.LeftButton}") && !_mouseState.leftButtonHeld ? 1 : 0;
+				fi.mouse.middleButtonPressed  = controller.IsPressed($"{Inputs.Mouse} {MouseInputs.MiddleButton}") && !_mouseState.middleButtonHeld ? 1 : 0;
+				fi.mouse.rightButtonPressed   = controller.IsPressed($"{Inputs.Mouse} {MouseInputs.RightButton}") && !_mouseState.rightButtonHeld ? 1 : 0;
+
+				// Button released criteria:
+				// If the input is not pressed in this frame and the button is held from before
+				fi.mouse.leftButtonReleased   = !controller.IsPressed($"{Inputs.Mouse} {MouseInputs.LeftButton}") && _mouseState.leftButtonHeld ? 1 : 0;
+				fi.mouse.middleButtonReleased = !controller.IsPressed($"{Inputs.Mouse} {MouseInputs.MiddleButton}") && _mouseState.middleButtonHeld ? 1 : 0;
+				fi.mouse.rightButtonReleased  = !controller.IsPressed($"{Inputs.Mouse} {MouseInputs.RightButton}") && _mouseState.rightButtonHeld ? 1 : 0;
+				fi.mouse.sensitivity = _syncSettings.MouseSensitivity;
+			}
+
+			// Updating mouse state
+			_mouseState.posX = fi.mouse.posX;
+			_mouseState.posY = fi.mouse.posY;
+			if (fi.mouse.leftButtonPressed    > 0) _mouseState.leftButtonHeld = true;
+			if (fi.mouse.middleButtonPressed  > 0) _mouseState.middleButtonHeld = true;
+			if (fi.mouse.rightButtonPressed   > 0) _mouseState.rightButtonHeld = true;
+			if (fi.mouse.leftButtonReleased   > 0) _mouseState.leftButtonHeld = false;
+			if (fi.mouse.middleButtonReleased > 0) _mouseState.middleButtonHeld = false;
+			if (fi.mouse.rightButtonReleased  > 0) _mouseState.rightButtonHeld = false;

```

Comment:
Idea for cleaning this up: create and populate a local `nextState` above and then assign it to `_mouseState` here

Reply:
Yes. [rebased](https://github.com/YoshiRulz/BizHawk_BranchBackup/commit/8b2e9f1130e974ff889cd9bb55037956ff3686e1)

Reply:
Pulled.